### PR TITLE
Save images from image tests

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: sunpy/tests/helpers.py [hunk 1]
```diff
@@ -77,16 +75,27 @@ def test_simple_plot():
...
         plt.figure()
-        name = "{0}.{1}".format(test_function.__module__, test_function.__name__)
         pngfile = tempfile.NamedTemporaryFile(delete=False)
         figure_hash = hash.hash_figure(test_function(*args, **kwargs), out_stream=pngfile)
         figure_test_pngfiles[name] = pngfile.name
         pngfile.close()
+
+        # Save the image that was generated
+        os.makedirs('result_images', exist_ok=True)
+        result_image_loc = 'result_images/{}.png'.format(name)
```
Comment (dpshelio) [inline sunpy/tests/helpers.py]: use `os.path.join` so it's valid for everyone.

File: sunpy/tests/helpers.py [hunk 2]
```diff
@@ -77,16 +75,27 @@ def test_simple_plot():
...
+                                test_function.__name__)
         plt.figure()
-        name = "{0}.{1}".format(test_function.__module__, test_function.__name__)
         pngfile = tempfile.NamedTemporaryFile(delete=False)
         figure_hash = hash.hash_figure(test_function(*args, **kwargs), out_stream=pngfile)
         figure_test_pngfiles[name] = pngfile.name
         pngfile.close()
+
+        # Save the image that was generated
+        os.makedirs('result_images', exist_ok=True)
```
Comment (dpshelio) [inline sunpy/tests/helpers.py]: I wonder whether this could be saved into a directory like where the `sunpy-figure-tests` repository would be.

Comment (Cadair) [reply sunpy/tests/helpers.py]: there should probably be a separate copy step, as this should also work on peoples computers etc.

**Comment by Cadair** [comment]
Are the test images not already saved to a temporary folder?

**Comment by nabobalis** [comment]
During the test? They are when I run them locally.

**Comment by dstansby** [comment]
I've overhauled the code a bit, so it only saves one copy and uses that for hashing. Lets see if the image tests pass...

**Comment by dstansby** [comment]
They do! 🎉

**Comment by nabobalis** [comment]
The only benefit I have had with the random directory is when I re-run the tests several times and have them all in seperate folders.

**Comment by Cadair** [comment]
Also, people won't commit it to the repo by mistake.

**Comment by Cadair** [comment]
What about having it in a local dir but one unique to each test run? to also fix @nabobalis point?

**Comment by dstansby** [comment]
👍 I've added a timestamp to the directory name, so (as long as you wait 1 second) running the tests a second time will dump the images in a new directory

File: sunpy/conftest.py [hunk 3]
```diff
@@ -47,17 +47,10 @@ def pytest_runtest_setup(item):
...
-            os.rename(figure_test_pngfiles[test_name], os.path.join(tempdir, test_name + '.png'))
-
+    if len(new_hash_library) > 0:
         # Write the new hash library in JSON
-        hashfile = os.path.join(tempdir, HASH_LIBRARY_NAME)
+        hashfile = os.path.join(test_fig_dir, HASH_LIBRARY_NAME)
         with open(hashfile, 'w') as outfile:
             json.dump(new_hash_library, outfile, sort_keys=True, indent=4, separators=(',', ': '))
 
-        print('All test files for figure hashes can be found in {0}'.format(tempdir))
```
Comment (nabobalis) [inline sunpy/conftest.py]: Do we not want this print statement somewhere else instead?

Comment (dstansby) [reply sunpy/conftest.py]: I've put a print statement back in.

File: sunpy/tests/helpers.py [hunk 4]
```diff
@@ -40,31 +41,30 @@
...
+test_fig_dir = 'result_images_{:%H%M%S}'.format(datetime.datetime.now())
 
 
 def figure_test(test_function):
     """
     A decorator for a test that verifies the hash of the current figure or the returned figure,
     with the name of the test function as the hash identifier in the library.
-    A PNG is also created with a temporary filename, with the lookup stored in the
-    `figure_test_pngfiles` dictionary.
+    A PNG is also created in the 'result_image' director, which is created
```
Comment (Cadair) [inline sunpy/tests/helpers.py]: directory

**Comment by Cadair** [comment]
ok so the conda env files really need updating: https://ci.appveyor.com/project/sunpy/sunpy/build/0.9.148#L125 why are we still trying to install the `wcsaxes` package?!

**Comment by Cadair** [comment]
@dstansby the appveyor fail is a "live one" that needs fixing, the docs fail looks to be our beloved VSO waving it's legs in the air.

**Comment by Cadair** [comment]
It's because circleci is uploading to dstansby/sunpy for some weird weird reason.

**Pull request merged**